gc3apps/ivs.uzh/gtree.py

```diff
@@ -0,0 +1,211 @@
+            self,
+            arguments = arguments,
+            inputs = inputs,
+            outputs = ["./results"],
+            stdout = 'gtree.log',
+            join=True,
+            executables = "./%s" % os.path.basename(input_function),
+            **extra_args)
+
+    def terminated(self):
+        """
+        Move results into original results folder
+        """
+        gc3libs.log.info("Application terminated with exit code %s" % self.execution.exitcode)
+        for result in os.listdir(os.path.join(self.output_dir,"results/")):
+            shutil.move(os.path.join(self.output_dir,"results/",result),
+                        os.path.join(self.input_folder,"results",os.path.basename(result)))
+        # Cleanup
+        os.removedirs(os.path.join(self.output_dir,"results/"))
+
+class GtreeScript(SessionBasedScript):
+    """
+    The script takes as input a rscript, the number of parallel jobs to run and the
+    dataset to be processed
+
+    The ``gtree`` command keeps a record of jobs (submitted, executed
+    and pending) in a session file (set name with the ``-s`` option); at
+    each invocation of the command, the status of all recorded jobs is
+    updated, output from finished jobs is collected, and a summary table
+    of all known jobs is printed.
+
+    Options can specify a maximum number of jobs that should be in
+    'SUBMITTED' or 'RUNNING' state; ``gtree`` will delay submission of
+    newly-created jobs so that this limit is never exceeded.
+    """
+
+    def __init__(self):
+        SessionBasedScript.__init__(
+            self,
+            version = __version__, # module version == script version
+            application = GtreeApplication,
+            # only display stats for the top-level policy objects
+            # (which correspond to the processed files) omit counting
+            # actual applications because their number varies over
+            # time as checkpointing and re-submission takes place.
+            stats_only_for = GtreeApplication,
+            )
+
+    def setup_args(self):
+        self.add_param('rscript', help=(
+            "Name of the R script to apply to the dataset."
+            "A corresponding `.R` file must exist in the current directory."))
+        self.add_param('P', type=int, help=("Value for the number of parallel executions"))
+        self.add_param('dataset', help=(
+            "Name of the dataset to be processed."))
+    # def parse_args(self):
+    #     """
+    #     Check that each element in args is at least a valid folder
+    #     """
+    #     for folder_name in self.params.args:
+    #         if not os.path.isdir(folder_name):
+    #             gc3libs.log.error(
+    #                 "Invalid input folder: {folder_name}."
+    #                 " Removing it from input list"
+    #                 .format(folder_name=folder_name))
+    #             self.params.args.remove(folder_name)
+
+    def get_function_name_and_file(self, rscript):
```

Comment:
You do not need this function here.  The function name is not a parameter that you're passing to the R script.